startups/views.py

```diff
@@ -207,37 +201,55 @@ def create(self, request):
             return Response(data, status=status.HTTP_201_CREATED)
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
-    # def update(self, request, pk=None):
-    #     # Implementation of PUT METHOD for one project - ExampLE URL: /api/projects/2/
-    #     # Do not forget about SLASH at the end of URL
-    #     # + you should send data in JSON
-    #     project_id = pk
-    #     project_updated_info = request.data
-    #     # ...
-    #     # PUT logic
-    #     # ...
-    #     data = {
-    #         'project_id': project_id,
-    #         'message': f"Hello, here's a PUT method! You update ALL information about PROJECT № {project_id}",
-    #         'updated_data': project_updated_info,
-    #         'status': status.HTTP_200_OK
-    #     }
-    #     return Response(data, status=status.HTTP_200_OK)
-    #
-    # def partial_update(self, request, pk=None):
-    #     # Implementation of PATCH METHOD for one project - ExampLE URL: /api/projects/2/
-    #     # Do not forget about SLASH at the end of URL
-    #     # + you should send data in JSON
-    #     # PATCHcing logic
-    #     project_id = pk
-    #     project_specific_updated_info = request.data
-    #     data = {
-    #         'project_id': project_id,
-    #         'message': f"Hello, here's a PATCH method! You update SOME information about project № {project_id}",
-    #         'specific_updated_data': project_specific_updated_info,
-    #         'status': status.HTTP_200_OK
-    #     }
-    #     return Response(data, status=status.HTTP_200_OK)
+    def update(self, request, pk=None):
+        """
+        Update an existing project with all fields required
+        (PUT api/projects/<pk>/)
+        Do not forget about SLASH at the end of URL
+        """
+        if not pk:
```

Comment:
in this line you might get "None", so going straight to "if" is not a good idea. Try wrapping it in a "try" block

Reply:
this class ProjectViewSet will be implemented in another issue

Reply:
thanks, added "try except" for this function